dts/bindings/ethernet/ethernet.yaml

```diff
@@ -12,3 +12,13 @@ properties:
       description: Specifies the MAC address that was assigned to the network device
     label:
       required: true
+    zephyr,random-mac-address:
+      type: boolean
+      required: false
+      description: |
+        Use a generated random MAC address dynamically on each boot.  Note that
```

Comment:
IMO this would be more clear as:
> Use a random MAC address generated when the driver is initialized.

Or other wording that makes extremely clear that the random address is not generated once then (dynamically re-)used on each boot.  (Also proposing changing "boot" in preparation for delayed driver initialization.)

This could also specify that the OUI remains the vendor OUI where one is available. That surprised me; if I were using this for privacy I might want the OUI randomized as well.

Reply:
Updated